crates/steel-core/src/parser/parser.rs

```diff
@@ -45,11 +45,20 @@ pub(crate) struct InterierSources {
 
 impl InterierSources {
     pub fn new() -> Self {
-        InterierSources {
+        let mut sources = InterierSources {
             paths: HashMap::new(),
             reverse: HashMap::new(),
             sources: Vec::new(),
-        }
+        };
+
+        // Here be dragons!
+        // In order to reduce the span size, we allocate
+        // a dummy source at the beginning. This means
+        // we don't need to use an Option<u32>, but rather
+        // just a plain old u32 for the source id.
+        sources.add_source("", None);
```

Comment:
This probably can just go back to being an `Option<u32>` after cleaning this up some more. I thought I was going to end up saving space but after working on it more I don't think that is the case.